go/v1beta1/storage/esutil/client.go

```diff
@@ -137,9 +146,32 @@ func (c *client) Create(ctx context.Context, request *CreateRequest) (string, er
 		indexOpts = append(indexOpts, c.esClient.Index.WithDocumentID(request.DocumentId))
 	}
 
+	var (
+		doc []byte
+		err error
+	)
+	if request.Join != nil {
+		doc, err = json.Marshal(&EsDocWithJoin{
```

Comment:
Looking at the PR as a whole, it makes sense that this will end up triggering `EsDocWithJoin. MarshalJSON`, but I could see coming in here later and incorrectly thinking that the protobuf message was being serialized with the wrong function. Maybe a comment here about the behavior?